pkg/cmd/util/clientcmd/factory_client_access.go

```diff
@@ -187,6 +189,39 @@ func (f *ring0Factory) UpdatePodSpecForObject(obj runtime.Object, fn func(*kapi.
 			template = &kapi.PodTemplateSpec{}
 		}
 		return true, fn(&template.Spec)
+	case *buildapi.BuildConfig:
+		kc, err := f.ClientSet()
+		if err != nil {
+			return false, err
+		}
+
+		labels.NewSelector()
+		pods, err := kc.Core().Pods(t.Namespace).List(metav1.ListOptions{LabelSelector: labels.SelectorFromSet(t.ObjectMeta.Labels).String()})
+		if err != nil {
+			return false, err
+		}
```

Comment:
i'm afraid this isn't right.  what needs to be updated is the buildconfig.spec.resources:
https://github.com/openshift/origin/blob/master/pkg/build/apis/build/v1/types.go#L71

Reply:
@bparees @csrwng thanks for the feedback, although I am not sure of an immediate approach to this as the function that gets passed in as an argument to this method receives a [`*PodSpec` type](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/set/set_resources.go#L193) as its only argument. Other types, such as deploymentconfigs have a `PodSpecTemplate` field as part of the Spec, but as far as I am aware, [buildconfigs do not](https://github.com/openshift/origin/blob/master/pkg/build/apis/build/types.go#L1006). 

An initial idea was to just pass a fake podSpec with a fake container, and a pointer to  buildConfig.Spec.Resources such that its value got mutated anyway, but unfortunately a podSpec.Containers[i].Resources receive values and not pointers.

Reply:
i'm thinking UpdatePodSpecForObject is probably just the wrong place to inject this logic.

Reply:
(for buildconfigs)

Reply:
@fabianofranz to solve this, maybe it'd be worth either:
  - adding a command wrapper in `oc` for `set_resources` that directly updates the `Resource` field on buildcofigs, or delegates to the upstream command for any other resource kinds

  - or, since we already have a downstream implementation of the `factory_client_access#UpdatePodSpecForObject`, modify the method signature on our end to receive an optional `*options` param with the desired value for the `Resources` field, which then gets set on this method as well (although not sure what we would then pass back to the caller of this method)